x/reporter/reporter.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+)
+
+var debugLog log.Logger = *log.New(io.Discard, "", 0)
+var httpClient = &http.Client{}
+
+func sendReport(record map[string]interface{}, collectorURL string) error {
```

Comment:
Please create a `Reporter` type that encapsulates the collector URL and sampling factors, so that a client can simply call `reporter.Report(report)`

Reply:
I made the changes to encapsulate the Record parameters. I was previously using the function name `sendReportRandomly` to be more descriptive but I switched to `Report` to be shorter.